src/core/pusher.ts

```diff
@@ -116,26 +132,46 @@ export default class Pusher {
           channel.handleEvent(event);
         }
       }
+
       // Emit globally [deprecated]
       if (!internal) {
         this.global_emitter.emit(event.event, event.data);
       }
     });
+
     this.connection.bind('connecting', () => {
       this.channels.disconnect();
     });
+
     this.connection.bind('disconnected', () => {
       this.channels.disconnect();
     });
+
     this.connection.bind('error', err => {
       Logger.warn(err);
     });
+  }
 
-    Pusher.instances.push(this);
-    this.timeline.info({ instances: Pusher.instances.length });
+  /**
+   * Allows you to switch Pusher cluster without
+   * losing all the channels/subscription binding
+   * as this is internally managed by the SDK.
+   */
+  switchCluster(options: ClusterOptions) {
```

Comment:
As you can see, we only allow to override the appKey and cluster, but of course this could be extend to the other config options but I guess this could be enabled once we have real usecases?

Reply:
Let's just expose appKey and cluster for now and enable other options when we have real use cases.